hooks/submitter_sgtk.py

```diff
@@ -0,0 +1,208 @@
+            )
+
+    def can_submit(self):
+        """
+        Checks if it's possible to submit versions given the current context/environment.
+
+        :returns:               Flag telling if the hook can submit a version.
+        :rtype:                 bool
+        """
+
+        if not self._upload_to_shotgun and not self._store_on_disk:
+            self.log_warning(
+                "App is not configured to store images on disk nor upload to shotgun!"
+            )
+
+            return False
+
+        return True
+
+    def submit_version(
+        self,
+        path_to_frames,
+        path_to_movie,
+        thumbnail_path,
+        sg_publishes,
+        sg_task,
+        description,
+        first_frame,
+        last_frame,
+    ):
+        """
+        Create a version in Shotgun for this path and linked to this publish.
+
+        :param path_to_frames: Path to the frame representation.
+        :param path_to_movie: Path to the movie representation.
+        :param thumbnail_path: Path to the thumbnail representing.
+        :param sg_publishes: Published files that have to be linked to the version.
+        :param sg_task: Task that have to be linked to the version.
+        :param description: Description of the version.
+        :param first_frame: Version first frame.
+        :param last_frame: Version last frame.
+
+        :returns:               Location of the rendered media
+        :rtype:                 str
+        """
+
+        # get current shotgun user
+        current_user = sgtk.util.get_current_user(self.__app.sgtk)
+
+        # create a name for the version based on the file name
+        # grab the file name, strip off extension
+        name = os.path.splitext(os.path.basename(path_to_movie))[0]
+        # do some replacements
+        name = name.replace("_", " ")
+        # and capitalize
+        name = name.capitalize()
+
+        # Create the version in Shotgun
+        ctx = self.__app.context
+        data = {
+            "code": name,
+            "sg_status_list": self.__app.get_setting("new_version_status"),
+            "entity": ctx.entity,
+            "sg_task": sg_task,
+            "sg_first_frame": first_frame,
+            "sg_last_frame": last_frame,
+            "sg_frames_have_slate": False,
+            "created_by": current_user,
+            "user": current_user,
+            "description": description,
+            "sg_path_to_frames": path_to_frames,
+            "sg_movie_has_slate": True,
+            "project": ctx.project,
+        }
+
+        if first_frame and last_frame:
+            data["frame_count"] = last_frame - first_frame + 1
+            data["frame_range"] = "%s-%s" % (first_frame, last_frame)
+
+        if sgtk.util.get_published_file_entity_type(self.__app.sgtk) == "PublishedFile":
+            data["published_files"] = sg_publishes
+        else:  # == "TankPublishedFile"
+            if len(sg_publishes) > 0:
```

Comment:
Why can't other publishes be registered when they are TankPublishedFile ? Where is this limitation coming from?

Reply:
I moved back old code. I don't know the reason behind this limitation.

Reply:
Bummer....